### PR TITLE
feat: scene room connection

### PR DESCRIPTION
**Pull request by NickKhalow**
- Room autoconnection based on the position
- Background execution

### CODE REVIEW

File: Explorer/Assets/DCL/Multiplayer/Connections/Archipelago/LiveConnections/AutoReconnectLiveConnection.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    {
+        private readonly IArchipelagoLiveConnection origin;
+        private readonly Action<string> log;
+        private string? cachedAdapterUrl;
+
+        public bool IsConnected => origin.IsConnected;
+
+        public AutoReconnectLiveConnection(IArchipelagoLiveConnection origin) : this(
+            origin,
+            m => Debug.Log($"{ReportCategory.ARCHIPELAGO_REQUEST}: {m}")
```
Comment (Kinerius) [inline Explorer/Assets/DCL/Multiplayer/Connections/Archipelago/LiveConnections/AutoReconnectLiveConnection.cs:21]: Shouldn't we use `ReportHub.Log`?

File: Explorer/Assets/DCL/Multiplayer/Connections/Archipelago/SignFlow/LogArchipelagoSignFlow.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+namespace DCL.Multiplayer.Connections.Archipelago.SignFlow
+{
+    public class LogArchipelagoSignFlow : IArchipelagoSignFlow
+    {
+        private const string PREFIX = "LogArchipelagoSignFlow:";
+        private readonly IArchipelagoSignFlow origin;
+        private readonly Action<string> log;
+
+        public LogArchipelagoSignFlow(IArchipelagoSignFlow origin) : this(origin, Debug.Log) { }
```
Comment (Kinerius) [inline Explorer/Assets/DCL/Multiplayer/Connections/Archipelago/SignFlow/LogArchipelagoSignFlow.cs:15]: `ReportHub.Log`?

**Review by Kinerius — COMMENTED** [review]
LGTM!

File: Explorer/Assets/DCL/Multiplayer/Connections/Archipelago/LiveConnections/AutoReconnectLiveConnection.cs (hunk 1 above)
Comment (NickKhalow) [reply Explorer/Assets/DCL/Multiplayer/Connections/Archipelago/LiveConnections/AutoReconnectLiveConnection.cs]: I change it to ReportHub.Log but for some reason it won't show the messages if the category is specified as ARCHIPELAGO_REQUEST, but it works if it's UNSPECIFIED

**Pull request merged**